### PR TITLE
Fix performance impacting race condition in deploy handler

### PR DESCRIPTION
**Pull request by arnaudsjs**
# Description

This PR fix the performance impact race condition, caused by the deploy handler, when a dependency is in the `deploying` state due to a repair run. This PR makes sure that the `requires` argument of the deploy handler can never contain a resource with the ResourceState "deploying".

closes inmanta/lsm#433

# Self Check:

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [x] Sufficient test cases (reproduces the bug/tests the requested feature)
- [x] Correct, in line with design
- [x] End user documentation is included or an issue is created for end-user documentation

# Reviewer Checklist:

- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Code is clear and sufficiently documented
- [ ] Correct, in line with design

### CODE REVIEW

File: tests/server/test_resourceservice.py [hunk 1]
```diff
@@ -261,3 +256,77 @@ async def test_events_resource_without_dependencies(server, client, environment,
...
+    await assert_status_fields(
+        r1_status=const.ResourceState.deployed,
+        r1_last_non_deploying_status=const.ResourceState.deployed,
+        r2_status=const.ResourceState.deploying,
+        r2_last_non_deploying_status=const.ResourceState.available,
+    )
+
+    # R1 start deployment + R2 skipped
+    action_id_r1 = await resource_deployer.start_deployment(rvid=rvid_r1_v1)
+    await resource_deployer.deployment_finished(rvid=rvid_r2_v1, action_id=action_id_r2, status=const.ResourceState.skipped)
```
Comment (wouterdb) [inline tests/server/test_resourceservice.py:316]: this test the new code path, but does this work on the old (resource_Action_update) code path as well?

**Review by wouterdb — COMMENTED** [review]
Looks nice

File: tests/server/test_resourceservice.py (hunk 1 above)
Comment (arnaudsjs) [reply tests/server/test_resourceservice.py]: Good remark. I updated the test case to verify both code paths.

File: tests/db/dumps/v202203160.sql [hunk 2]
```diff
@@ -0,0 +1,992 @@
...
+
+-- Dumped from database version 13.4
+-- Dumped by pg_dump version 13.4
+
+SET statement_timeout = 0;
+SET lock_timeout = 0;
+SET idle_in_transaction_session_timeout = 0;
+SET client_encoding = 'UTF8';
+SET standard_conforming_strings = on;
+SELECT pg_catalog.set_config('search_path', '', false);
```
Comment (sanderr) [inline tests/db/dumps/v202203160.sql:13]:
```suggestion
--SELECT pg_catalog.set_config('search_path', '', false);
```

File: tests/db/test_v202111260_to_v202203160.py [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+from db.common import PGRestore
+from inmanta.const import ResourceState
+from inmanta.data import Resource
+from inmanta.server.bootloader import InmantaBootloader
+
+
+@pytest.fixture
+async def migrate_v202111260_to_v202203160(
+    hard_clean_db, hard_clean_db_post, postgresql_client: Connection, server_config
+) -> AsyncIterator[Callable[[], Awaitable[None]]]:
```
Comment (sanderr) [inline tests/db/test_v202111260_to_v202203160.py:33]: Did you know there is now the `migrate_db_from` fixture defined in `conftest.py`?

**Review by sanderr — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]
Forgot I had requested a change (the db dump one).

**Review by sanderr — APPROVED** [review]

**Pull request closed without merging**